dask/dataframe/core.py

```diff
@@ -4461,9 +4461,12 @@ def hash_shard(df, nparts, split_out_setup=None, split_out_setup_kwargs=None):
         h = split_out_setup(df, **(split_out_setup_kwargs or {}))
     else:
         h = df
-    h = hash_pandas_object(h, index=False)
+    from .utils import hash_object_dispatch
+
+    hash_object_func = hash_object_dispatch.dispatch(type(h))
```

Comment:
I think you can just call `has_object_dispatch(h.values, ...)`. You shouldn't need the `.dispatch`.

Reply:
I think it has to instead be `has_object_dispatch(h, ...)` and not `has_object_dispatch(h.values, ...)` right? Assuming `h.values` to be a typo since `h` could be anything like `Index, Series, Dataframe` who could have indexes too to be hashed based on `index` flag.